includes/acl/core/memory.h

```diff
@@ -405,4 +408,19 @@ namespace acl
 			src_bit_offset += num_bits_copied;
 		}
 	}
+
+	//////////////////////////////////////////////////////////////////////////
+
+	// TODO: get an official L3 cache size
+	constexpr size_t CACHE_FLUSH_BUFFER_BYTES = 12 * 1024 * 1024;
+
+	inline void flush_data_cache(Allocator& allocator)
+	{
+		uint8_t* buffer = allocate_type_array<uint8_t>(allocator, CACHE_FLUSH_BUFFER_BYTES);
+
+		for (size_t i = 0; i < CACHE_FLUSH_BUFFER_BYTES; ++i)
+			buffer[std::rand() % CACHE_FLUSH_BUFFER_BYTES] ^= buffer[std::rand() % CACHE_FLUSH_BUFFER_BYTES];
```

Comment:
not guaranteed to touch every cache line :( also no need for random access, we just want to flush the cache, ideally we want to do this with SIMD linearly so the processor can prefetch ahead and help make this somewhat fast. there is no reason why it should be slower than it needs to be